src/python/pants/option/parser.py

```diff
@@ -597,7 +605,9 @@ def expand(val_or_str):
         def merge_in_rank(vals):
             if not vals:
                 return None
-            expanded_vals = [to_value_type(expand(x)) for x in vals]
+            expanded_vals = [to_value_type(v) for v in (expand(i) for i in vals) if v is not None]
```

Comment:
This is close to getting to be a little too complex for comprehension IMO, but still ended on it for the sake of conciseness.